### ISSUE TITLE
magicsock: do active discovery of peers

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#483]
Tracking bug for redoing how peer path discovery works.

New plan:

* add a new `DiscoKey` discovery key type (https://github.com/tailscale/tailscale/commit/88c305c8afa15dbee5a6cb8b98e5547cd19f4e7e)
* generate a new discovery key pair whenever a node is starting up
* shared the discovery key through control to peers (it's separate from the wireguard keypair)
* magicsock can then send encrypted UDP packets saying "Are you there?" and "Yes I am" (as needed, based on peer activity), and probably PMTU later
* stop telling Wireguard-go that we even have N peer routes. Just make up one dummy one and let magicsock completely own the path discovery

**Comment by bradfitz** [comment]
This is done as of 0.100.0-0, now out in stable for Linux, Windows, macOS, and approved but not released for iOS. (we were waiting on the beta track approvals to let people choose which track they're running, but that approval hasn't arrived yet)
